File: app/src/main/java/com/geekbrains/kotlin_lessons/sharedPeferences/SharedPreferencesManager.java
package com.geekbrains.kotlin_lessons.sharedPeferences;

import android.app.Activity;
import android.content.SharedPreferences;

public class SharedPreferencesManager {

    private final SharedPreferences sPreferences;

    private SharedPreferences.Editor sEditor;

    public SharedPreferencesManager(Activity activity){
        sPreferences= activity.getPreferences(Activity.MODE_PRIVATE);
    }

    public SharedPreferences.Editor getEditor() {
        return sPreferences.edit();
    }


    public void storeBoolean(String tag, boolean value) {
        sEditor = getEditor();
        sEditor.putBoolean(tag, value);
        sEditor.commit();
    }

    public void storeString(String tag, String str) {
        sEditor = getEditor();
        sEditor.putString(tag, str);
        sEditor.commit();
    }

    public boolean retrieveBoolean(String tag, boolean defValue) {
        return sPreferences.getBoolean(tag, defValue);

    }


    public String retrieveString(String tag, String defStr) {
        return sPreferences.getString(tag, defStr);
    }


    public int retrieveInt(String tag, int defValue) {
        return sPreferences.getInt(tag, defValue);
    }


    public void storeInt(String tag, int defValue) {
        sEditor = getEditor();
        sEditor.putInt(tag, defValue);
        sEditor.commit();
    }


}

File: app/src/main/java/com/geekbrains/kotlin_lessons/sharedPeferences/SharedPreferencesManager.kt
package com.geekbrains.kotlin_lessons.sharedPeferences

import android.app.Activity
import android.content.SharedPreferences
import android.content.SharedPreferences.Editor

class SharedPreferencesManager(activity: Activity) {
    private val sPreferences: SharedPreferences = activity.getPreferences(Activity.MODE_PRIVATE)
    private var sEditor: Editor? = null
    private val editor: Editor
        get() = sPreferences.edit()

    fun storeBoolean(tag: String?, value: Boolean) {
        sEditor = editor
        sEditor!!.putBoolean(tag, value)
        sEditor!!.commit()
    }

    fun storeString(tag: String?, str: String?) {
        sEditor = editor
        sEditor!!.putString(tag, str)
        sEditor!!.commit()
    }

    fun retrieveBoolean(tag: String?, defValue: Boolean): Boolean {
        return sPreferences.getBoolean(tag, defValue)
    }

    fun retrieveString(tag: String?, defStr: String?): String? {
        return sPreferences.getString(tag, defStr)
    }

    fun retrieveInt(tag: String?, defValue: Int): Int {
        return sPreferences.getInt(tag, defValue)
    }

    fun storeInt(tag: String?, defValue: Int) {
        sEditor = editor
        sEditor!!.putInt(tag, defValue)
        sEditor!!.commit()
    }

}